packages/components/src/Tabs/hooks/useArrowKeyNavigation.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from "react";
+
+interface UseArrowKeyNavigationProps {
+  elementsRef: React.RefObject<(HTMLElement | null)[]>;
+  onActivate: (index: number) => void;
+}
+
+export const useArrowKeyNavigation = ({
+  elementsRef,
+  onActivate,
+}: UseArrowKeyNavigationProps) => {
+  const handleKeyDown = useCallback(
```

Comment:
Good call out to see what other DS are doing! I do have a working generic hook but I could see it growing and yes, it's a valid concern that it might just be more complicated.  I'll dive into this one as side desk work, do some research and we can regroup on it